lib/bamboo/mailer.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Bamboo.Mailer do
+  defmacro __using__(opts) do
+    quote bind_quoted: [opts: opts] do
+      %{adapter: adapter, config: config} = Bamboo.Mailer.parse_opts(__MODULE__, opts)
+
+      @adapter adapter
+      @config config
+
+      def deliver(email) do
+        Bamboo.Mailer.deliver(@adapter, email, @config)
+      end
+    end
+  end
+
+  def deliver(_, %{from: nil}, config) do
```

Comment:
Would this make sense as an adapter implemented error instead of on `mailer` itself?